apps/payroll/contracts/Payroll.sol

```diff
@@ -0,0 +1,586 @@
+     * @return Integer with the monthly amount
+     */
+    function calculatePayrollBurnrate() public constant auth(OWNER_READ_ROLE) returns(uint256 payrollBurnrate) {
+        payrollBurnrate = yearlyTotalPayroll / 12;
+    }
+
+    /**
+     * @dev Days until the contract can run out of funds
+     * @notice Days until the contract can run out of funds
+     * @return Integer with the number of days
+     */
+    function calculatePayrollRunway() public constant auth(OWNER_READ_ROLE) returns(uint256 payrollRunway) {
+        if (yearlyTotalPayroll == 0)
+            payrollRunway = 2**256 - 1;
+        else
+            payrollRunway = this.balance.mul(365) / yearlyTotalPayroll;
+    }
+
+    /* EMPLOYEE ONLY */
+    /**
+     * @dev Set token distribution for payments to an employee (the caller)
+     * @notice Set token distribution for payments to an employee (the caller).
+     *         Only callable once every 6 months
+     * @param tokens Array with the tokens to receive, they must belong to allowed tokens for employee
+     * @param distribution Array (correlated to tokens) with the proportions (integers over 100)
+     */
+    function determineAllocation(address[] tokens, uint256[] distribution) public auth(EMPLOYEE_ROLE) {
+        var employee = employees[employeeIds[msg.sender]];
+        require(getTimestamp() > employee.lastAllocation && getTimestamp() - employee.lastAllocation > 15768000); // half a year in seconds
+
+        // check arrays match
+        require(tokens.length == distribution.length);
+
+        // check distribution is right
+        uint256 sum = 0;
+        uint256 i;
+        for (i = 0; i < distribution.length; i++) {
+            // check token is allowed
+            require(employee.allowedTokens[tokens[i]]);
+            // set distribution
+            employee.allocation[tokens[i]] = distribution[i];
+            sum = sum.add(distribution[i]);
+        }
+        require(sum == 100);
+
+        employee.lastAllocation = getTimestamp();
+    }
+
+    /**
+     * @dev Get payment proportion for a token and an employee (the caller)
+     * @notice Get payment proportion for a token and an employee (the caller)
+     * @param token The token address
+     */
+    function getAllocation(address token) public constant auth(EMPLOYEE_ROLE) returns(uint256 allocation) {
+        var employee = employees[employeeIds[msg.sender]];
+        allocation = employee.allocation[token];
+    }
+
+    /**
+     * @dev payday To withdraw monthly payment by employee (the caller)
+     * @notice Only callable once a month. Assumed token has these standard checks implemented:
+     *         https://theethereum.wiki/w/index.php/ERC20_Token_Standard#How_Does_A_Token_Contract_Work.3F
+     */
+    function payday() public auth(EMPLOYEE_ROLE) {
+        var employee = employees[employeeIds[msg.sender]];
+        // check that employee exists (and matches)
+        require(employee.accountAddress == msg.sender);
```

Comment:
with this check, we can remove the `auth(EMPLOYEE_ROLE)` completely

Reply:
Certainly